File: src/main/java/offer/technical/test/model/UserResource.java
package offer.technical.test.model;

import java.time.LocalDate;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.Data;
import offer.technical.test.model.validation.AdultConstraint;
import offer.technical.test.model.validation.CountryConstraint;
import offer.technical.test.model.validation.GenderConstraint;
import offer.technical.test.model.validation.PhoneNumberConstraint;

@Data
public class UserResource {

  @NotBlank
  private String name;

  @NotNull
  @AdultConstraint
  private LocalDate birthDate;

  @NotBlank
  @CountryConstraint
  private String country;

  @PhoneNumberConstraint
  private String phoneNumber;

  @GenderConstraint
  private String gender;

}

File: src/main/kotlin/offer/technical/test/model/UserResource.kt
package offer.technical.test.model

import offer.technical.test.model.validation.AdultConstraint
import offer.technical.test.model.validation.CountryConstraint
import offer.technical.test.model.validation.GenderConstraint
import offer.technical.test.model.validation.PhoneConstraint
import java.time.LocalDate
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class UserResource (

    @field:NotBlank
    val name: String,

    @field:AdultConstraint
    @field:NotNull
    val birthDate: LocalDate,

    @field:CountryConstraint
    @field:NotBlank
    val country: String,

    @field:PhoneConstraint
    val phoneNumber: String,

    @field:GenderConstraint
    val gender: String

)